From 93e0d5d2433ac10f92aaa22bc280750059bb7fee Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Wed, 4 Nov 2020 17:40:13 +0100
Subject: [PATCH] Improve font config deserialization

Going from QString -> QVariant -> QFont isn't ideal and means that it tries
to create a QFont from a null/empty QString as well, resulting in a warning:
QFont::fromString: Invalid description '(empty)'
Just skip the QVariant step and call QFont::fromString directly.
---
 src/greeter/GreeterApp.cpp | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/greeter/GreeterApp.cpp b/src/greeter/GreeterApp.cpp
index bbce512..01f53fa 100644
--- a/src/greeter/GreeterApp.cpp
+++ b/src/greeter/GreeterApp.cpp
@@ -258,10 +258,13 @@ namespace SDDM {
         }
 
         // Set font
-        QVariant fontEntry = mainConfig.Theme.Font.get();
-        QFont font = fontEntry.value<QFont>();
-        if (!fontEntry.toString().isEmpty())
-            QGuiApplication::setFont(font);
+        const QString fontStr = mainConfig.Theme.Font.get();
+        if (!fontStr.isEmpty()) {
+            QFont font;
+            if (font.fromString(fontStr)) {
+                QGuiApplication::setFont(font);
+            }
+        }
 
         // Set session model on proxy
         m_proxy->setSessionModel(m_sessionModel);
-- 
2.30.0

